libiocage/lib/Host.py

```diff
@@ -63,7 +63,7 @@ def __init__(
             host=self,
             logger=self.logger
         )
-        self._defaults = defaults
+        self._defaults = defaults if defaults is not None else {}
```

Comment:
Got your point! This should achieve the same without setting _defaults to a property other than None or DefaultResource.

```python
class HostGenerator:

    _class_distribution = libiocage.lib.Distribution.DistributionGenerator

    _devfs: 'libiocage.lib.DevfsRules.DevfsRules'
    _defaults: 'libiocage.lib.Resource.DefaultResource'
    releases_dataset: libzfs.ZFSDataset

    def __init__(
        self,
        root_dataset: libzfs.ZFSDataset=None,
        defaults: 'libiocage.lib.Resource.DefaultResource'=None,
        zfs: 'libiocage.lib.ZFS.ZFS'=None,
        logger: 'libiocage.lib.Logger.Logger'=None
    ) -> None:

        self.logger = libiocage.lib.helpers.init_logger(self, logger)
        self.zfs = libiocage.lib.helpers.init_zfs(self, zfs)

        self.datasets = libiocage.lib.Datasets.Datasets(
            root=root_dataset,
            logger=self.logger,
            zfs=self.zfs
        )
        self.distribution = self._class_distribution(
            host=self,
            logger=self.logger
        )

        if defaults is not None:
            self._defaults = defaults
```